src/client/auth.rs

```diff
@@ -148,6 +149,25 @@ impl AuthMechanism {
 
                 Ok(())
             }
+            #[cfg(feature = "gssapi-auth")]
+            AuthMechanism::Gssapi => {
+                if credential.username.is_none() {
+                    return Err(ErrorKind::InvalidArgument {
+                        message: "No username provided for GSSAPI authentication".to_string(),
+                    }
+                    .into());
+                }
+
+                if credential.source.as_deref().unwrap_or("$external") != "$external" {
+                    return Err(ErrorKind::InvalidArgument {
```

Comment:
Hmm, just noticed that the "must be $external" message for OIDC is slightly different than everything else, maybe we should update it 🤔

Reply:
Updated 👍